subprojects/publish/src/main/java/org/gradle/api/publish/internal/PublicationInternal.java

```diff
@@ -28,6 +29,9 @@
 
     ModuleVersionIdentifier getCoordinates();
 
+    @Nullable
```

Comment:
Why make this nullable if implementations could return `ImmutableAttributes.EMPTY`? Unless `null` has a different semantics than an empty attribute set (which is not the case right?).

Reply:
done